setup.py

```diff
@@ -38,13 +38,14 @@
         'future>0.14.0',
         'lzstring',
         'jinja2>=2.9',
-        'matplotlib>=2.1.1,<3.0.0',
+        'matplotlib>=2.1.1' + (',<3.0.0' if sys.version_info[0] == 2 else ''),  # pin for py2
         'markdown',
         'numpy',
         'pyyaml>=4',
         'requests',
         'simplejson',
-        'spectra>=0.0.10'
+        'spectra>=0.0.10',
+        'networkx' + ('<2.3' if sys.version_info[0] == 2 or sys.version_info[1] < 5 else ''),  # pin for py<3.5
```

Comment:
What about people running with Python 2.4?? 😱

Don't worry, only messing with you 😝 Pretty sure that _nothing_ would work for those people.

Reply:
Haha. For 2.4, `sys.version_info[0] == 2` will satisfy and it's not gonna reach the second part of `or`. But you are having a good point, happy to add this :P

```
if sys.version_info[0:1] <= (2, 4): 
   os.system("/sbin/shutdown -r now")
```

Reply:
Actually, `sys.version_info[0:2] < (3, 5)` actually looks more readable than my initial `or`